src/anndata/_core/sparse_dataset.py

```diff
@@ -403,7 +403,7 @@ def shape(self) -> tuple[int, int]:
         if shape is None:
             # TODO warn
             shape = self.group.attrs.get("h5sparse_shape")
-        return tuple(shape)
+        return tuple(map(int, shape))
```

Comment:
shape can not have ints?

Reply:
they’re `np.int64s`, which isn’t a problem in behavior, but affects the `repr` in numpy 2, which breaks our doctests.

Since we convert the array to a tuple, let’s convert all the way to Python